pkgs/games/unvanquished/default.nix

```diff
@@ -0,0 +1,156 @@
+{ lib, stdenv, writeScript, fetchzip, fetchurl, fetchgit, buildFHSUserEnv
+, runCommandNoCC, makeDesktopItem, copyDesktopItems, gcc, cmake, gmp, libGL
+, zlib, ncurses, geoip, lua5, nettle, curl, SDL2, freetype, glew, openal
+, libopus, opusfile, libogg, libvorbis, libjpeg, libwebp, libpng
+, libtheora # TODO: drop once we get to 0.52
+, cacert, aria2 # to download assets
+}:
+
+let
+  version = "0.51.2";
+  binary-deps-version = "4";
+
+  unvanquished-binary-deps = stdenv.mkDerivation rec {
+    # DISCLAIMER: this is selected binary crap from the NaCl SDK
+    name = "unvanquished-binary-deps";
+    version = binary-deps-version;
+    src = fetchzip {
+      url = "https://dl.unvanquished.net/deps/linux64-${version}.tar.bz2";
+      sha256 = "08bpyavbh5lmyprvqqi59gnm8s1fjmlk9f1785wlv7f52d9f9z1p";
+    };
+    dontPatchELF = true;
+    preFixup = ''
+      # We are not using the autoPatchelfHook, because it would make
+      # nacl_bootstrap_helper unable to load nacl_loader:
+      # "nacl_loader: ELF file has unreasonable e_phnum=13"
+      interpreter="$(< "$NIX_CC/nix-support/dynamic-linker")"
+      for f in pnacl/bin/*; do
+        if [ -f "$f" && -x "$f" ]; then
+          echo "Patching $f"
+          patchelf --set-interpreter "$interpreter" "$f"
+        fi
+      done
+    '';
+    preCheck = "pnacl/bin/clang -v"; # check it links correctly
+    installPhase = ''
+      mkdir -p $out
+      cp -R ./* $out/
+    '';
+  };
+
+  preload = stdenv.mkDerivation {
+    name = "nix-preload-for-unvanquished-nacl";
+    buildCommand = ''
+      mkdir $out/etc -p
+      echo ${gcc.cc.lib}/lib/libstdc++.so.6 > $out/etc/ld-nix.so.preload
+    '';
+  };
+
+  fhsEnv = buildFHSUserEnv {
+    name = "unvanquished-fhs-wrapper";
+    targetPkgs = pkgs: [ gcc.cc.lib preload ];
+  };
+
+  unvanquished-assets = stdenv.mkDerivation {
+    pname = "unvanquished-assets";
+    inherit version;
+
+    src = fetchurl {
+      # master as of 2021-04-09: the 0.51.2 script isn't as great
+      url = "https://raw.githubusercontent.com/Unvanquished/Unvanquished/2783e77acc2e41fc1f717124394ed851f1b612e6/download-paks";
```

Comment:
How about unifying this with the Unvanquished sources ?

e.g.
```
unvanquished-sources = pkgs.fetchFromGitHub {
    owner = "Unvanquished";
    repo = "Unvanquished";
    rev = "v${version}";
    hash="sha256-1psegPxm2O8iBJIYpxjdauMLv5MWZvn0MtqbimgtF+o=";
};
[...]
src = "${unvanquished-sources}/download-paks";
```

Reply:
Unfortunately the download-paks script didn't exist in 0.51.1. There was another one that has been rewritten since.